src/suggestions.ts

```diff
@@ -50,38 +77,48 @@ function makeSuggestion(queryName, element, content, {variant, name}) {
       if (warning) {
         console.warn(warning)
       }
-      let [text, options] = queryArgs
+      const [text, options] = queryArgs
 
-      text = typeof text === 'string' ? `'${text}'` : text
+      const normalizedText = typeof text === 'string' ? `'${text}'` : text
 
-      options = options
+      const stringifiedOptions = options
         ? `, { ${Object.entries(options)
-            .map(([k, v]) => `${k}: ${v}`)
+            .filter<[string, boolean | RegExp]>(isDefinedOption)
+            .map(([k, v]) => `${k}: ${v.toString()}`)
             .join(', ')} }`
         : ''
 
-      return `${queryMethod}(${text}${options})`
+      return `${queryMethod}(${normalizedText.toString()}${stringifiedOptions})`
     },
   }
 }
 
-function canSuggest(currentMethod, requestedMethod, data) {
+function canSuggest(
+  currentMethod: string,
+  requestedMethod: string | undefined,
+  data: string | null,
+) {
   return (
     data &&
     (!requestedMethod ||
       requestedMethod.toLowerCase() === currentMethod.toLowerCase())
   )
 }
 
-export function getSuggestedQuery(element, variant = 'get', method) {
+export function getSuggestedQuery(
+  element: Element,
+  variant: Variant = 'get',
+  method?: Method,
+): Suggestion | undefined {
   // don't create suggestions for script and style elements
   if (element.matches(DEFAULT_IGNORE_TAGS)) {
     return undefined
   }
 
   //We prefer to suggest something else if the role is generic
-  const role =
-    element.getAttribute('role') ?? getImplicitAriaRoles(element)?.[0]
+  const role: string =
+    element.getAttribute('role') ??
+    (getImplicitAriaRoles(element) as string[])[0]
```

Comment:
From what are you casting here?

Reply:
as for screen `getImplicitAriaRoles` has not been converted so it will return an array of any. Maybe we can add a TODO for those two cases?

Reply:
I have migrated role-helpers so we will not have this cast anymore